lib/matplotlib/cbook.py

```diff
@@ -2428,6 +2428,107 @@ def safe_first_element(obj):
     return next(iter(obj))
 
 
+def normalize_kwargs(kw, alias_mapping=None, required=(), forbidden=(),
+                     allowed=None):
+    """Helper function to normalize kwarg inputs
+
+    The order they are resolved are:
+
+     1. aliasing
+     2. required
+     3. forbidden
+     4. allowed
+
+    This order means that only the canonical names need appear in
+    `allowed`, `forbidden`, `required `
+
+    Parameters
+    ----------
+
+    alias_mapping, dict, optional
+        A mapping between a canonical name to a list of
+        aliases, in order of precedence from lowest to highest.
+
+        If the canonical value is not in the list it is assumed to have
+        the highest priority.
+
+    required : iterable, optional
+        A tuple of fields that must be in kwargs.
+
+    forbidden : iterable, optional
+        A list of keys which may not be in kwargs
+
+    allowed : tuple, optional
+        A tuple of allowed fields.  If this non None, then raise if
```

Comment:
On 2016/01/03 10:43 AM, Thomas A Caswell wrote:
> I think 'non' is what I intended, is 'not' clearer?
Yes, it is.  "non" is not a standalone word, as far as I know.